File: STP/src/gestionAplicacion/Personas/Persona.java
package gestionAplicacion.Personas;
import java.io.Serializable;

public abstract class Persona implements Serializable{
    protected String nombre;
    protected String documento;
    protected String direccion;
    protected String edad;

    public Persona(String nombre, String documento, String direccion, String edad) {
        this.nombre = nombre;
        this.documento = documento;
        this.direccion = direccion;
        this.edad = edad;
    }
    
    public String presentacion() {
    	return "Nombre: " + this.nombre + ". Edad: " + this.edad + " años "; 
    }
    public Persona() {
    	super();
    }
    
    public String getNombre() {
        return nombre;
    }

    public String getDocumento() {
        return documento;
    }

    public String getDireccion() {
        return direccion;
    }

    public String getEdad() {
        return edad;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setDocumento(String documento) {
        this.documento = documento;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public void setEdad(String edad) {
        this.edad = edad;
    }
    
    
}

File: STP-Python/src/gestionAplicacion/Personas/Persona.py
from abc import ABC

class Persona(ABC):
    def __init__(self, *args):
        if len(args) == 4:
            self._nombre = args[0]
            self._documento = args[1]
            self._direccion = args[2]
            self._edad = args[3]
        
        elif len(args)==0:
            self._nombre = None
            self._documento = None
            self._direccion = None
            self._edad = None


    def getNombre(self):
        return self._nombre

    def setNombre(self, nombre):
        self._nombre = nombre

    def getDocumento(self):
        return self._documento
    
    def setDocumento(self,documento):
        self._documento = documento

    def getDireccion(self):
        return self._direccion

    def setDireccion(self,direccion):
        self._direccion = direccion

    def getEdad(self):
        return self._edad

    def setEdad(self,edad):
        self._edad = edad
